gateway/api/connections/helpers.go

```diff
@@ -83,6 +83,45 @@ func coerceToAnyMap(src map[string]string) map[string]any {
 	return dst
 }
 
+func validatePatchConnectionRequest(req openapi.ConnectionPatch) error {
```

Comment:
Here we have the same validations as the function below. It might be better to have a single validation function shared by both.

Reply:
We have this new function to handle possible `nil` fields.